packages/rum-core/src/domain/vital/vitalCollection.ts

```diff
@@ -0,0 +1,63 @@
+import type { ClocksState } from '@datadog/browser-core'
+import { elapsed, generateUUID } from '@datadog/browser-core'
+import { LifeCycleEventType } from '../lifeCycle'
+import type { LifeCycle, RawRumEventCollectedData } from '../lifeCycle'
+import type { RawRumVitalEvent } from '../../rawRumEvent.types'
+import { RumEventType, VitalType } from '../../rawRumEvent.types'
+
+export interface DurationVitalStart {
+  name: string
+  startClocks: ClocksState
+}
+
+export interface DurationVitalStop {
+  name: string
+  stopClocks: ClocksState
+}
+
+interface Vital {
+  name: string
+  type: VitalType
+  startClocks: ClocksState
+  value: number
+}
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏We could explicitly specify an `DurationVital` type
```suggestion
interface DurationVital {
  name: string
  type: VitalType.Duration
  startClocks: ClocksState
  value: Duration
}
```
Wdyt?

Reply:
I had some thoughts about how much distinction we should make now between Vital, CustomVital, CustomDurationVital.
I settled on decoupling the `processVital` from the exact type of Vital (without handling the logic around custom VS non-custom just yet) which is similar to what we are doing for action for example.

But yes, we could remove some future thinking and just go with what is really handled for now.